src/migrate/v0.4.md

```diff
@@ -0,0 +1,398 @@
+[#1344]: https://github.com/godot-rust/gdext/pull/1344
+[#1346]: https://github.com/godot-rust/gdext/pull/1346
+
+[api-callable]: https://godot-rust.github.io/docs/gdext/master/godot/builtin/struct.Callable.html
+
+
+### `Singleton` trait
+
+[#1325]: Singleton access has been moved to a dedicated [`Singleton`][api-singleton] trait, enabling generic programming while maintaining
+backward compatibility with existing `singleton()` methods.
+
+In the future, we may look into other options to provide singletons, especially once we support them for user classes, too.
+
+**Before (0.3):**
+
+```rust
+let input = Input::singleton();
+let engine = Engine::singleton();
+```
+
+**After (0.4):**
+
+If you already include the prelude, no code changes should be necessary:
+
+```rust
+use godot::prelude::Singleton;
+
+let input = Input::singleton(); // Still works
+let engine = Engine::singleton(); // Still works
+
+// Now supports generic access:
+fn summon_the_one<T: Singleton>() -> Gd<T> {
+    T::singleton()
+}
+```
+
+[#1325]: https://github.com/godot-rust/gdext/pull/1325
+
+[api-singleton]: https://godot-rust.github.io/docs/gdext/master/godot/obj/trait.Singleton.html
+
+
+### Removed conversions
+
+[#1286], [#1316]: By-value `From` conversions between string types have been removed to clarify that no buffer optimization occurs during conversion.
+You can still use `From` for references.
+
+**Before (0.3):**
+
+```rust
+let gstring = GString::from("hello");
+let sname = StringName::from(gstring); // or .into()
+```
+
+**After (0.4):**
+
+```rust
+let gstring = GString::from("hello");
+let sname = StringName::from(&gstring);
+```
+
+The `From<&'static CStr>` implementation for `StringName` has been removed, since Godot no longer offers a performance benefit,
+and static storage duration may very easily cause memory leaks.
+
+Furthermore, `impl From<VariantArray> for Packed*Array` has been removed. This was unsound; `VariantArray` can contain arbitrary types.
+
+[#1286]: https://github.com/godot-rust/gdext/pull/1286
+[#1316]: https://github.com/godot-rust/gdext/pull/1316
+
+
+### Renames
+
+- Struct `ClassName` -> [`ClassId`][api-classid], which represents the lightweight idea of IDs better.
+- Method `apply_deferred()` -> [`run_deferred()`][api-gd-rundeferred] and [`run_deferred_gd()`][api-gd-rundeferredgd],
+  accepting both `&mut T` and `Gd<T>` parameters.
+- Method `Base::to_gd()` -> [`Base::to_init_gd()`][api-base-toinitgd], for proper access during `init`.
+- Method `Callable::from_local_fn()` -> [`Callable::from_fn()`][api-callable-fromfn] (see above).
+- Assoc fn `Callable::from_local_static()` -> [`Callable::from_class_static()`][api-callable-fromclassstatic].
+- Trait `ToSignalObject` -> [`ObjectToOwned`][api-objecttoowned].
+- Trait `WithDeferredCall` -> merged into [`WithBaseField`][api-withbasefield].
+
+[api-classid]: https://godot-rust.github.io/docs/gdext/master/godot/obj/struct.ClassId.html
```

Comment:
returns 404, should be https://godot-rust.github.io/docs/gdext/master/godot/meta/struct.ClassId.html